pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -180,3 +184,90 @@ type HTTPSDConfig struct {
+	// Consul Datacenter name, If not provided it will use the local Consul Agent Datacenter.
+	// +optional
+	Datacenter *string `json:"datacenter,omitempty"`
+	// Namespaces are only supported in Consul Enterprise.
+	// +optional
+	Namespace *string `json:"namespace,omitempty"`
+	// Admin Partitions are only supported in Consul Enterprise.
+	// +optional
+	Partition *string `json:"partition,omitempty"`
+	// HTTP Scheme default "http"
+	// +kubebuilder:validation:Enum=http;https
+	// +optional
+	Scheme *string `json:"scheme,omitempty"`
+	// A list of services for which targets are retrieved. If omitted, all services are scraped.
+	// +listType:=atomic
+	// +optional
+	Services []string `json:"services,omitempty"`
+	// An optional list of tags used to filter nodes for a given service. Services must contain all tags in the list.
+	//+listType:=atomic
+	// +optional
+	Tags []string `json:"tags,omitempty"`
+	// The string by which Consul tags are joined into the tag label.
+	// If unset, Prometheus uses its default value.
+	// +optional
+	TagSeparator *string `json:"tag_separator,omitempty"`
+	// Node metadata key/value pairs to filter nodes for a given service.
+	// +mapType:=atomic
+	// +optional
+	NodeMeta map[string]string `json:"node_meta,omitempty"`
+	// Allow stale Consul results (see https://www.consul.io/api/features/consistency.html). Will reduce load on Consul.
+	// If unset, Prometheus uses its default value.
+	// +optional
+	AllowStale *bool `json:"allow_stale,omitempty"`
+	// The time after which the provided names are refreshed.
+	// On large setup it might be a good idea to increase this value because the catalog will change all the time.
+	// If unset, Prometheus uses its default value.
+	// +optional
+	RefreshInterval *v1.Duration `json:"refresh_interval,omitempty"`
+	// BasicAuth information to authenticate against the Consul Server.
+	// More info: https://prometheus.io/docs/operating/configuration/#endpoints
+	// +optional
+	BasicAuth *v1.BasicAuth `json:"basicAuth,omitempty"`
+	// Authorization header configuration to authenticate against the Consul Server.
+	// +optional
+	Authorization *v1.SafeAuthorization `json:"authorization,omitempty"`
+	// Optional OAuth 2.0 configuration.
+	// +optional
+	Oauth2 *v1.OAuth2 `json:"oauth2,omitempty"`
+	// Optional proxy URL.
+	// +optional
+	ProxyUrl *string `json:"proxy_url,omitempty"`
+	// Comma-separated string that can contain IPs, CIDR notation, domain names
+	// that should be excluded from proxying. IP and domain names can
+	// contain port numbers.
+	// +optional
+	NoProxy *string `json:"no_proxy,omitempty"`
+	// Use proxy URL indicated by environment variables (HTTP_PROXY, https_proxy, HTTPs_PROXY, https_proxy, and no_proxy)
+	// If unset, Prometheus uses its default value.
+	// +optional
+	ProxyFromEnvironment *bool `json:"proxy_from_environment,omitempty"`
+	// Specifies headers to send to proxies during CONNECT requests.
+	// +optional
```

Comment:
(nit)
```suggestion
	// +mapType:=atomic
	// +optional
```

Reply:
Out of my curiosity, why we don't need optional here?

Reply:
I only suggested to add `+mapType:=atomic` but optional is still needed :)